src/app/core/utils/meta-reducers.ts

```diff
@@ -13,6 +14,25 @@ export function resetOnLogoutMeta<S>(reducer: ActionReducer<S>): ActionReducer<S
   };
 }
 
+export function resetPersonalizedShoppingMeta(reducer: ActionReducer<ShoppingState>): ActionReducer<ShoppingState> {
```

Comment:
This is very specific to the shopping store, so maybe it should be declared there.

Alternatively, you could generalize it a bit, something like `resetSubStatesOnLogoutMeta<T>(subStates: keyOf T): ((reducer: ActionReducer<T>) => ActionReducer<T>)`, that uses `lodash-es` `omit` on the state.